src/test/java/spoon/reflect/ast/AstCheckerTest.java

```diff
@@ -158,6 +161,18 @@ private boolean isNotCandidate(CtMethod<?> candidate) {
 					|| notCandidates.contains(candidate.getDeclaringType().getSimpleName() + "#" + candidate.getSimpleName());
 		}
 
+		private boolean hasPrimitiveTypeForParameters(List<CtParameter<?>> parameters) {
```

Comment:
I don't see the relation with this PR. Why was it working before?

Reply:
I'm not sure. I suspect that on almost all other setters we were not using primitive types. I can extract it in another PR if you want: the relation here is that it fix a broken test resulting by the new setter.

Reply:
Bizarre. We have plenty of other setters with boolean (eg setImplicit, setShadow, etc). It would be much better to understand what happens here.

Reply:
I'll have a look

Reply:
Good thing you insisted on that one: I forgot to call the model change listener... I updated the contract here because it was not clear enough.